app/src/main/java/bg/zahov/app/ui/workout/WorkoutScreen.kt

```diff
@@ -0,0 +1,768 @@
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    style = MaterialTheme.typography.labelLarge
+                )
+            } else {
+                Icon(
+                    modifier = Modifier.size(24.dp),
+                    painter = painterResource(R.drawable.ic_settings_dots_blue),
+                    tint = MaterialTheme.colorScheme.onTertiaryContainer,
+                    contentDescription = ""
+                )
+            }
+        }
+        DropdownMenu(
+            expanded = isDropDownExpanded,
+            onDismissRequest = onClose
+        ) {
+            when (itemType) {
+                ItemType.EXERCISE -> {
+                    ExerciseMenuItem.entries.toList().forEachIndexed { index, item ->
+                        DropdownMenuItem(text = {
+                            Text(
+                                text = stringResource(item.stringResource),
+                                maxLines = 1,
+                                overflow = TextOverflow.Ellipsis,
+                                style = MaterialTheme.typography.labelLarge
+                            )
+                        },
+                            onClick = {
+                                when (item) {
+                                    ExerciseMenuItem.ADD_NOTE -> onFirstOption()
+                                    ExerciseMenuItem.REPLACE -> onSecondOption()
+                                    ExerciseMenuItem.REMOVE -> onThirdOption()
+                                }
+                                onClose()
+                            })
+                    }
+                }
+
+                ItemType.SET -> {
+                    SetMenuItem.entries.toList().forEachIndexed { index, item ->
+                        DropdownMenuItem(text = {
+                            Text(
+                                text = stringResource(item.stringResource),
+                                maxLines = 1,
+                                overflow = TextOverflow.Ellipsis,
+                                style = MaterialTheme.typography.labelLarge
+                            )
+                        },
+                            onClick = {
+                                when (item) {
+                                    SetMenuItem.WARMUP -> onFirstOption()
+                                    SetMenuItem.DROP_SET -> onSecondOption()
+                                    SetMenuItem.FAILURE -> onThirdOption()
+                                }
+                                onClose()
+                            })
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Enum class representing the available set menu items for a workout.
+ * Each enum constant corresponds to a specific set type.
+ *
+ * @property stringResource The string resource ID associated with the set type name.
+ */
```

Comment:
Probably extract in different file

Reply:
👍 Created a new file within the whole 'workout' package because another screen will need these